test/test_mps.py

```diff
@@ -9250,8 +9250,68 @@ def weight_int8pack_mm(a, b_int8pack, b_scales):
             self.assertLess(mean_err, 0.05)
 
 
+class TestSDPA(TestCaseMPS):
+    def _compare_tensors(self, y, ref):
+        denom = torch.maximum(ref.abs(), torch.tensor([1e-6], device=ref.device, dtype=ref.dtype))
+        err = ((y - ref).abs() / denom).mean().item()
+        self.assertLess(err, 0.01)
 
+    def _test_sdpa_no_mask(self, is_causal: bool, dtype: torch.dtype, L: int = 1, S: int = 72):
+        torch.manual_seed(1729)
+        with torch.nn.attention.sdpa_kernel([torch.nn.attention.SDPBackend.MATH]):
+            q = torch.randn([1, 32, L, 128], dtype=dtype, device="mps")
```

Comment:
also test with different sizes like 23, 121. Not just power of 2.

Reply:
done!